source/routing/tests/routing_api_test.cpp

```diff
@@ -2744,3 +2800,585 @@ BOOST_AUTO_TEST_CASE(section_geometry_with_shapes) {
+    section = journey.sections(1);
+    BOOST_CHECK_EQUAL(section.type(), pbnavitia::SectionType::PUBLIC_TRANSPORT);
+    BOOST_CHECK_EQUAL(section.origin().stop_point().name(), "stop_point:sa1:s1");
+    BOOST_CHECK_EQUAL(section.destination().stop_point().name(), "stop_point:sa3:s1");
+    BOOST_CHECK_EQUAL(section.begin_date_time(), "20180309T080400"_pts);
+    BOOST_CHECK_EQUAL(section.end_date_time(), "20180309T082300"_pts);
+}
+
+namespace {
+
+    nr::NightBusFilter::Params get_default_filter_params() {
+        return  nr::NightBusFilter::Params {
+                    navitia::DateTimeUtils::set(0, "08:00"_t),
+                    true,
+                    nr::NightBusFilter::default_max_factor,
+                    nr::NightBusFilter::default_base_factor};
+    }
+
+    std::tuple<nr::Journey, nr::Journey> build_night_bus_journeys(ed::builder & b) {
+
+        b.vj("vj1")
+            ("A1", "09:00"_t, "09:00"_t)("C1", "12:00"_t, "12:00"_t);
+
+        b.vj("vj2")
+            ("B1", "22:00"_t, "22:00"_t)("C1", "23:00"_t, "23:00"_t);
+
+        nt::PT_Data & d = *b.data->pt_data;
+        auto stl1 = d.vehicle_journeys[0]->stop_time_list;
+        auto stl2 = d.vehicle_journeys[1]->stop_time_list;
+
+        nr::Journey j1;
+        j1.sections.push_back(nr::Journey::Section(
+            stl1[0], navitia::DateTimeUtils::set(0, "09:00"_t),
+            stl1[1], navitia::DateTimeUtils::set(0, "10:00"_t)
+        ));
+        j1.departure_dt = navitia::DateTimeUtils::set(0, "08:30"_t);
+        j1.arrival_dt = navitia::DateTimeUtils::set(0, "10:30"_t);
+
+        nr::Journey j2;
+        j2.sections.push_back(nr::Journey::Section(
+            stl2[0], navitia::DateTimeUtils::set(0, "22:00"_t),
+            stl2[1], navitia::DateTimeUtils::set(0, "23:00"_t)
+        ));
+        j2.departure_dt = navitia::DateTimeUtils::set(0, "21:55"_t);
+        j2.arrival_dt = navitia::DateTimeUtils::set(0, "23:10"_t);
+
+        return std::make_tuple(j1, j2);
+    }
+}
+
+BOOST_AUTO_TEST_CASE(night_bus_should_treated_as_way_later)
+{
+    ed::builder b("20120614");
+
+    nr::Journey j1, j2;
+    std::tie(j1, j2) = build_night_bus_journeys(b);
+
+    nr::NightBusFilter::Params filter_params = get_default_filter_params();
+
+    BOOST_CHECK_EQUAL(nr::way_later(j1, j2, filter_params), false);
+    BOOST_CHECK_EQUAL(nr::way_later(j2, j1, filter_params), true);
+}
+
+BOOST_AUTO_TEST_CASE(night_bus_should_treated_as_way_later_anti_clockwise)
+{
+    ed::builder b("20120614");
+
+    nr::Journey j1, j2;
+    std::tie(j1, j2) = build_night_bus_journeys(b);
+
+    nr::NightBusFilter::Params filter_params = get_default_filter_params();
+    filter_params.requested_datetime = navitia::DateTimeUtils::set(0, "23:30"_t);
+    filter_params.clockwise = false;
+
+    BOOST_CHECK_EQUAL(nr::way_later(j1, j2, filter_params), true);
+    BOOST_CHECK_EQUAL(nr::way_later(j2, j1, filter_params), false);
+}
+
+BOOST_AUTO_TEST_CASE(should_filter_night_bus_when_too_late)
```

Comment:
The test is more `night_bus_filter_is_order_agnostic` ?

Reply:
:heavy_check_mark: